libs/partners/google-vertexai/langchain_google_vertexai/llms.py

```diff
@@ -0,0 +1,471 @@
+from __future__ import annotations
+
+from concurrent.futures import Executor, ThreadPoolExecutor
+from typing import Any, ClassVar, Dict, Iterator, List, Optional, Union
+
+import vertexai  # type: ignore
+from google.api_core.client_options import ClientOptions
+from google.cloud.aiplatform.gapic import (
+    PredictionServiceAsyncClient,
+    PredictionServiceClient,
+)
+from google.cloud.aiplatform.models import Prediction
+from google.protobuf import json_format
+from google.protobuf.struct_pb2 import Value
+from langchain_core.callbacks.manager import (
+    AsyncCallbackManagerForLLMRun,
+    CallbackManagerForLLMRun,
+)
+from langchain_core.language_models.llms import BaseLLM
+from langchain_core.outputs import Generation, GenerationChunk, LLMResult
+from langchain_core.pydantic_v1 import BaseModel, Field, root_validator
+from vertexai.language_models import (  # type: ignore
+    CodeGenerationModel,
+    TextGenerationModel,
+)
+from vertexai.language_models._language_models import (  # type: ignore
+    TextGenerationResponse,
+)
+from vertexai.preview.generative_models import GenerativeModel, Image  # type: ignore
+from vertexai.preview.language_models import (  # type: ignore
+    CodeGenerationModel as PreviewCodeGenerationModel,
+)
+from vertexai.preview.language_models import (
+    TextGenerationModel as PreviewTextGenerationModel,
+)
+
+from langchain_google_vertexai._utils import (
+    create_retry_decorator,
+    get_client_info,
+    is_codey_model,
+    is_gemini_model,
+)
+
+
+def _completion_with_retry(
+    llm: VertexAI,
+    prompt: List[Union[str, Image]],
+    stream: bool = False,
+    is_gemini: bool = False,
+    run_manager: Optional[CallbackManagerForLLMRun] = None,
+    **kwargs: Any,
+) -> Any:
+    """Use tenacity to retry the completion call."""
+    retry_decorator = create_retry_decorator(run_manager=run_manager)
```

Comment:
I think we might need to plumb `max_retries` through here in order to make it retry, unless there's a way to configure retries in the latest vertexai sdk?

Reply:
done